src/storage/cpu_shared_storage_manager.h

```diff
@@ -215,12 +215,12 @@ void CPUSharedStorageManager::FreeImpl(const Storage::Handle& handle) {
       << "Failed to unmap shared memory. munmap failed with error "
       << strerror(errno);
 
+  if (count == 0) {
 #ifdef __linux__
-  CHECK_EQ(close(handle.shared_id), 0)
-      << "Failed to close shared memory. close failed with error "
-      << strerror(errno);
+    CHECK_EQ(close(handle.shared_id), 0)
```

Comment:
if(count==0) shouldnt' be moved. Skip close if shared_id == -1 instead

Reply:
You mean close on first FreeImpl and skip if already closed?